### PR TITLE
Simplify invalidateOnFileChange() in Transformers

### PR DESCRIPTION
**Pull request by grishy**
# ↪️ Pull Request

In [docs](https://parceljs.org/plugin-system/transformer/#DependencyOptions)/code:

```
invalidateOnFileChange(FilePath): void
```

You don't need to add await before

## ✔️ PR Todo

- [ ] Added/updated unit tests for this change
- [ ] Filled out test instructions (In case there aren't any unit tests)
- [ ] Included links to related issues/PRs

### CODE REVIEW

**Comment by mischnic** [comment]
There's actually one more case here which could just be a for loop without any Promises: https://github.com/parcel-bundler/parcel/blob/d58e3361ad16c90bb44a2cad8edae31df66c7578/packages/transformers/posthtml/src/PostHTMLTransformer.js#L108-L115

**Comment by grishy** [comment]
Oh, I see. I will update it, if you want in this PR.

File: packages/transformers/posthtml/src/PostHTMLTransformer.js [hunk 1]
```diff
@@ -104,15 +104,10 @@ export default (new Transformer({
...
-            return asset.invalidateOnFileChange(filePath);
-          }
-          return Promise.resolve();
-        }),
-      );
+    for (let {type, file: filePath} of res.messages) {
+      if (type === 'dependency') {
+        asset.invalidateOnFileChange(filePath);
+      }
     }
```
Comment (mischnic) [inline packages/transformers/posthtml/src/PostHTMLTransformer.js]: Let's keep the `if`, not sure if `messages` might be unset sometimes
```suggestion
    if (res.messages) {
      for (let {type, file: filePath} of res.messages) {
        if (type === 'dependency') {
          asset.invalidateOnFileChange(filePath);
        }
      }
    }
```

**Review by mischnic — APPROVED** [review]
Thanks!

**Pull request merged**